kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -450,13 +508,23 @@ func (c *ipamController) onBlockUpdated(kvp model.KVPair) {
 			}
 			if len(c.allocationsByNode[node]) == 0 {
 				delete(c.allocationsByNode, node)
+				clearNodeFromPoolCounters(node)
```

Comment:
I am not 100% sure this is the right time to clear the reclaimed IP count - just because there are no allocations doesn't mean the node doesn't exist, or even that the blocks affine to that node are gone.

Seems like we should keep incrementing the counter for a given node until it's actually deleted from the k8s API, or almost equivalently there are no IPAM _blocks_ assigned to the node (a subtle different from there being no allocations)

Reply:
Thanks for raising this. Do you have a preference between node counter cleanup occurring on K8S deletion vs. the last block release?

K8S deletion would be slightly cleaner and perhaps more aligned with user expectation?

Reply:
I think doing it when the node no longer exists is the right time. 

We do already check for the existence of k8s nodes here: https://github.com/projectcalico/calico/blob/f5de453524d6ec89c877e904f9e1cedf0aa14c9d/kube-controllers/pkg/controllers/node/ipam.go#L859

So you could probably drive this off of that already existing check.